packages/dapp/src/components/Navbar/Menu/WalletMenu/WalletMenu.tsx

```diff
@@ -113,11 +113,13 @@ export const WalletMenu = ({ handleClose }: NavbarMenuProps) => {
     onWalletDisconnect();
   };
 
-  const handleMultisigEnvironmentCheck = async () => {
+  const handleMultisigEnvironmentCheck = useCallback(async () => {
     const response = await checkMultisigEnvironment();
 
     setIsMultisigEnvironment(response);
-  };
+    // Check MultisigEnvironment only on first render
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
@Abhikumar98 Could you please take a look and maybe test the multisig-wallet check? Thanks ahead

Reply:
Hey @dennyscode,

This is working as expected and tbh, I don't think `useCallback` is required here. In both scenarios, with and without `useCallback` the code is working fine, although I'd suggest to revert back to using without it unless there is an issue being thrown.